src/services/orderService.ts

```diff
@@ -1,43 +1,128 @@
-import { findOrdersByStatus,createOrder, createOrderItems, findCartItemsByUserId, clearCartForUser } from "../reposetories/orderRepository";
-
+import { findOrdersByStatus,createOrder, findCartItemsByUserId, clearCartForUser } from "../reposetories/orderRepository";
+import { createOrderItems } from '../reposetories/orderItemRepository';
+import { getDiscountById } from '../reposetories/discountRepository';
+import { getProductByIdRepository, updateProductRepository } from '../reposetories/productRepository';
+import { deleteCartItemsByUserId, getCartItemsWithProductDetails } from '../reposetories/cartItemReposirtory';
+import { getOrderItemsService } from "../services/orderItemService";
 export const placeOrderService = async (userId: string, couponId: string, status: string) => {
   // Step 1: Fetch cart items
-  const cartItems = await findCartItemsByUserId(userId);
+  const cartItems = await getCartItemsWithProductDetails(userId);
 
   if (cartItems.length === 0) {
-    throw new Error("Cart is empty");
+      throw new Error("Cart is empty");
+  }
+
+  // Step 2: Validate stock availability for all cart items
+  for (const cartItem of cartItems) {
+      const product = await getProductByIdRepository(cartItem.productId); // Fetch product details
```

Comment:
I think this approach is better instead of using a loop:  

```typescript
const productIds = cartItems.map(item => item.productId); 
const products = await Promise.all(
  productIds.map(async productId => {
    const product = await getProductByIdRepository(productId);
    if (!product) {
      throw new Error(`Product not found for ID: ${productId}`);
    }
    return product;
  })
);
```

Reply:
and you can Validate stock for all products by use loop